base/bucket.go

```diff
@@ -132,6 +132,9 @@ type BucketSpec struct {
 	MaxConcurrentQueryOps         *int           // maximum number of concurrent query operations (default: DefaultMaxConcurrentQueryOps)
 	BucketOpTimeout               *time.Duration // How long bucket ops should block returning "operation timed out". If nil, uses GoCB default.  GoCB buckets only.
 	KvPoolSize                    int            // gocb kv_pool_size - number of pipelines per node. Initialized on GetGoCBConnString
+	KvBufferSize                  int
+	DcpBuffer                     int
+	Serverless                    bool
```

Comment:
I don't feel like 'Serverless' should be part of the bucket spec.  It looks like it's only here so that it can be used to set the DcpBuffer, but we've added DcpBuffer here too.